lighting-estimation-explainer.md

```diff
@@ -0,0 +1,91 @@
+
+As lighting is scene-relative as opposed to display-relative, the luminance values are encoded linearly with no gamma curve.  Most modern render engines perform intermediate calculations in linear space and can accept such values directly.  If an engine performs intermediate calculations in a color space encoded with gamma, such as SRGB, care must be taken when converting the values.  After scaling the values, the result may include components above 1.0 or below 0.0.  Naive implementations that clamp RGB components independently will result in erraneous hue and saturation for out-of-gamut colors.
+
+## Global Illumination
+
+Rendering algorithms take into consideration not only the light received by a surface from the light source but also light that has bounced around the scene multiple times before reaching the eye.
+
+Traditional real-time engines had a simple global "ambient" constant value that is added to the real-time shading result.  Engines using such a simple technique can use XRLightProbe.indirectIrradiance, scaled to return the desired effect.  It may also be necessary to apply a gamma curve if the shading is done in SRGB space.
+
+Global illumination describes the collective techniques used to more accurately estimate the light received from indirect reflections.
+
+## Image based lighting
+
+HDR Cube Map textures, as created by the XRReflectionProbe provide all the information about light sources and indirect bounces needed to accurately render PBR materials that are diffuse, glossy, and visibly reflective.  Image based lighting effects utilizing such textures are simple to implement and perform well for VR and AR rendering.  Unfortunately, such cube map textures require a lot of video memory and can often represent the environment from a limited range of locations where such a map was captured.
+
+HDR Cube Map textures are commonly used to implement "Reflection Probes" in modern rendering engines.
+
+## Spherical Harmonics
+
+SH (Spherical Harmonics) are used as a more compact alternative to HDR cube maps by storing a small number of coefficient values describing a fourier series over the surface of a sphere.  SH can effectively compress cube maps, while retaining multiple lights and directionality.  Due to their lightweight nature, many SH probes can be used within a scene, be interpolated, or be calculated for locations nearer to the lit objects.
+
+WebXR API supports up to 9 SH coefficients per RGB color component, for a total of 27 floating point scalar values.  This enables the level 2 (3rd) order of details.  If a platform can not supply all 9 coefficients, it can pass 0 for the higher order coefficients resulting in an effectively lower frequency reproduction.  
+
+This "SH probe" format is used by most modern rendering engines, including Unity, Unreal, and Threejs.
+
+## Shadows
+
+When an HDR Cube Map texture is available, shadows only have to consider occlusion of other rendered objects in the scene.
```

Comment:
Doing this 100% correctly is very hard; however, there are many simple approximations that are commonly used.  In particular, it may be sufficient in some cases to combine a baked ambient occlusion map with a IBR shader that needs no realtime dynamic lights.

A simple, non-physically based implementation may simply index the HDR cube map using a surface normal, and blend it with an albedo term representing the color of the surface using operators representing the artists intent.